### PR TITLE
Feat/submit ama

### PR DESCRIPTION
**Pull request by stephanernst**
Can successfully create an AMA. Fixed a number of bugs.
Quadra programmed.

### CODE REVIEW

**Comment by AboudS777** [comment]
Awesome work Stephane! Now we can proceed to allow users to add comments/questions, like the AMA and vote on comments/AMA. :)

File: src/main/java/ama/account/UserController.java [hunk 1]
```diff
@@ -50,16 +46,11 @@ public String login(Model model) {
 
     @PostMapping("/login")
     public String loggedIn(Model model) {
-        return "account";
-    }
-
-    @GetMapping("/registered")
-    public String registered(Model model) {
-        return "login";
+        return "redirect:/account";
```
Comment (RyanBillard) [inline src/main/java/ama/account/UserController.java]: 👍

File: src/main/java/ama/post/CommentPost.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Created by Stephane on 2017-03-19.
+ */
+@Entity
+public class CommentPost extends Post {
+
+    @ManyToOne
+    private final Post context;
+
+    public CommentPost() {this.context = null;}
```
Comment (RyanBillard) [inline src/main/java/ama/post/CommentPost.java]: why are we initializing to null? What is the context used for?

File: src/main/java/ama/post/CommentPost.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public CommentPost(User op, Post context, String text) {
+        super(op, text);
+        this.context = context;
+    }
+
+    public Post getContext() {
+        return this.context;
+    }
+
+    /*  don't think we need this anymore
```
Comment (RyanBillard) [inline src/main/java/ama/post/CommentPost.java]: delete this if not needed pls

File: src/main/java/ama/post/PostController.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @Autowired
+    private SubmissionPostRepository submissionPostRepository;
+
+    @Autowired
+    private Validator validator;
+
+    @GetMapping("/create_submission")
+    public String createAMASubmission(Model model) {
+        model.addAttribute("submissionPost",new SubmissionPost());
+        return "postsubmission";
```
Comment (RyanBillard) [inline src/main/java/ama/post/PostController.java]: can we make 'postsubmission' and 'create_submission' the same string for consistency's sake?

File: src/main/java/ama/post/PostController.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        if(result.hasErrors()) {
+            return "postsubmission";
+        }
+        submissionPostRepository.save(post);
+        return "redirect:/";
+    }
+
+    @GetMapping("/posts/{submission}")
+    public String getSubmissionView(@PathVariable(value = "submission") String submission, Model model) {
+        SubmissionPost post = submissionPostRepository.findByTitle(submission);
```
Comment (RyanBillard) [inline src/main/java/ama/post/PostController.java]: what's the title here? is it a string that could have spaces? If so that could be a problem

Comment (RyanBillard) [inline src/main/java/ama/post/PostController.java]: maybe we can change to the ID

File: src/main/java/tests/AuthenticationTest.java [hunk 6]
```diff
@@ -26,44 +26,34 @@
...
+    private MultiValueMap<String,Object> user;
+
     @Before
     public void setUp() throws Exception {
-
         this.base = new URL("http://localhost:" + port + "/");
-
+        user = new LinkedMultiValueMap<>();
+        user.add("username","sarran");
+        user.add("password","theman");
```
Comment (RyanBillard) [inline src/main/java/tests/AuthenticationTest.java]: 😎

File: src/main/java/ama/post/CommentPost.java (hunk 2 above)
Comment (SarrankanPathmanathan) [inline src/main/java/ama/post/CommentPost.java]: context is the post to which the comment is directed to. The context can either be the parent post, or a comment within that post.

File: src/main/java/ama/post/PostController.java (hunk 4 above)
Comment (stephanernst) [inline src/main/java/ama/post/PostController.java]: Sure thing fam.

File: src/main/java/ama/post/CommentPost.java (hunk 2 above)
Comment (stephanernst) [inline src/main/java/ama/post/CommentPost.java]: Had it as a final variable but going to change it so we don't need to initialize to null anymore.

File: src/main/java/ama/post/CommentPost.java (hunk 3 above)
Comment (stephanernst) [inline src/main/java/ama/post/CommentPost.java]: but what if we need it?

File: src/main/java/ama/post/PostController.java (hunk 5 above)
Comment (stephanernst) [inline src/main/java/ama/post/PostController.java]: The string can't have spaces. Spaces get filled with "%20" for urls.

Comment (RyanBillard) [inline src/main/java/ama/post/PostController.java]: nice

**Comment by RyanBillard** [comment]
LGTM GL HF GG NO RE

**Pull request merged**